### PR TITLE
fix(coverage): don't type check

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit changes "deno coverage" command not to type check.

Instead of relying on infrastructure for module loading in "deno run";
the code now directly reaches into cache for original and transpiled
sources. In case sources are not available the error is returned to the
user, suggesting to first run "deno test --coverage" command.

Fixes https://github.com/denoland/deno/issues/13161
Fixes https://github.com/denoland/deno/issues/13326

### CODE REVIEW

File: cli/tools/coverage.rs [hunk 1]
```diff
@@ -643,40 +643,65 @@ pub async fn cover_files(
...
-    let script_source = &module_source.code;
 
+    let maybe_file = if module_specifier.scheme() == "file" {
+      ps.file_fetcher.get_source(&module_specifier)
+    } else {
+      ps.file_fetcher
+        .fetch_cached(&module_specifier, 10)
+        .with_context(|| {
+          format!("Failed to fetch \"{}\" from cache.", module_specifier)
+        })?
```
Comment (bartlomieju) [inline cli/tools/coverage.rs:654]: This is fishy to me; we shouldn't collect coverage data for remote scripts. I need to double check it.

**Comment by bartlomieju** [comment]
I verified that this PR works with `deno_std`.

File: cli/tools/coverage.rs [hunk 2]
```diff
@@ -643,40 +643,66 @@ pub async fn cover_files(
...
+
+    // Check if file was transpiled
+    let transpiled_source = match file.media_type {
+      MediaType::JavaScript
+      | MediaType::Unknown
+      | MediaType::Cjs
+      | MediaType::Mjs
+      | MediaType::Json => file.source.as_ref().clone(),
+      MediaType::Dts => "".to_string(),
+      _ => {
```
Comment (dsherret) [inline cli/tools/coverage.rs:675]: Perhaps we should remove this `_ =>` and instead have all the possibilities here? I kind of feel like we need a simpler enum that we can easily get from a `MediaType` that only has stuff like `JavaScript, `Typescript`, `Declaration`, etc in it.

File: cli/tools/coverage.rs [hunk 3]
```diff
@@ -643,40 +643,66 @@ pub async fn cover_files(
...
+    };
+
+    // Check if file was transpiled
+    let transpiled_source = match file.media_type {
+      MediaType::JavaScript
+      | MediaType::Unknown
+      | MediaType::Cjs
+      | MediaType::Mjs
+      | MediaType::Json => file.source.as_ref().clone(),
+      MediaType::Dts => "".to_string(),
```
Comment (dsherret) [inline cli/tools/coverage.rs:674]: What about `Dmts` and `Dcts`?

File: cli/tools/coverage.rs [hunk 4]
```diff
@@ -643,40 +643,66 @@ pub async fn cover_files(
...
+    let maybe_file = if module_specifier.scheme() == "file" {
+      ps.file_fetcher.get_source(&module_specifier)
+    } else {
+      ps.file_fetcher
+        .fetch_cached(&module_specifier, 10)
+        .with_context(|| {
+          format!("Failed to fetch \"{}\" from cache.", module_specifier)
+        })?
+    };
+    let file = if let Some(f) = maybe_file {
```
Comment (dsherret) [inline cli/tools/coverage.rs:656]: `maybe_file.ok_or_else(|| ...)?;`?

**Review by dsherret — COMMENTED** [review]
Just needs the media type fix.

File: cli/tools/coverage.rs (hunk 2 above)
Comment (bartlomieju) [reply cli/tools/coverage.rs:675]: Good point. I've taken this code directly from `cli/proc_state.rs` I guess it should be updated there as well

File: cli/tools/coverage.rs (hunk 3 above)
Comment (bartlomieju) [reply cli/tools/coverage.rs:674]: See comment above

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**